### DISCUSSION TITLE
Do  you have video tutorial on how to do negative keyword/Asin using bulk file?

### DISCUSSION THREAD

**Discussion by news2023** [amzn/ads-advanced-tools-docs#71] [Q&A] [answered]
Do  you have video tutorial on how to do negative keyword/Asin using bulk file?

**Comment by DrRockett** [comment, marked as answer]
Hi, thanks for reaching out! We do not have a video specific to negative targets, but I do have a few resources/guidelines for you. 

- First, [this section of an article about creating Sponsored Products campaigns](http://-%20i%20have%20a%20personal%20conflict%20this%20afternoon/) lists the fields required to create negative targets. ([Here is the same information for Sponsored Brands](https://advertising.amazon.com/API/docs/en-us/bulksheets/2-0/create-sb-campaign#optional-add-more-entities-to-the-campaign)). 
- Next, if you are managing Sponsored Products campaigns, there is a very helpful feature we recently released called [Input guidance for Sponsored Products](https://advertising.amazon.com/API/docs/en-us/bulksheets/2-0/input-guidance). You can turn on this feature in the checklist when you download a custom spreadsheet on the bulk ops main page. After you select values for the first three columns (Product, Entity, Operation), the required and optional fields will be highlighted. So if you select “Negative keyword” for the Entity, you will see the necessary fields that you need to create that entity. 
- Finally, I’m attaching an example file that I hope will help. The file has two examples in the same tab: One new campaign where I’ve added negative targets, and one existing campaign where I’m adding new negative targets to the existing campaign/ad group. You can see the input guidance in the example sheet as well.



I hope this information is helpful, but please comment back if you need more information. 

[Negative_targets_bulksheets_example.xlsx](https://github.com/amzn/ads-advanced-tools-docs/files/11214479/Negative_targets_bulksheets_example.xlsx)

**Reply by news2023** [reply]
I really appreciate your response, and the sample bulk sheet is really helpful! I will use it as my reference.

**Comment by news2023** [comment]
I have a follow up question, under OPERATION - if I didn't put any changes like update, create, etc. - what will be the possible result or will there be a possible changes that may happen to may ads?

**Reply by DrRockett** [reply]
Hi! Very good question--if you **leave the operation field blank** in a particular row, there would be no changes made to that row when you upload the file. This allows you to download a bulksheet that might have hundreds or thousands of rows, and just make a couple of updates without needing to delete other rows, and without overloading our system when you upload the file. All of the rows with no operation would be ignored, so it makes the processing more efficient. 

Another thing I wanted to point out, which you can see in the sample file I shared, is that for **new** campaigns and ad groups, you need to put a temporary ID. You'll put that exact same temporary ID in all of the child entities as well. Then, once the campaign is uploaded/created, those IDs will become the unique IDs that you see in the downloaded file. [This video tutorial](https://advertising.amazon.com/API/docs/en-us/bulksheets/2-0/bulksheets-videos#how-to-create-sponsored-products-auto-campaigns) will show all of what I am describing in this comment. There are several videos on that page that might also be useful as well. Hope this helps!

**Reply by news2023** [reply]
That's awesome! Thank you much for the reference, big help!

**Comment by news2023** [comment]
"Also, I notice that there is no option to delete a campaign on the Ad console. The only option I see here is to archive it. After doing some research, I found that I can use a bulk sheet operation and put "DELETE" to remove the campaign. Is this the correct process?"

**Reply by DrRockett** [reply]
Hey there, the three options are CREATE, UPDATE, and ARCHIVE. Archiving is essentially the same thing as deleting (although it's not truly deleted/removed, an archived campaign cannot be "un-archived," so it will not start running again). [This article section](https://advertising.amazon.com/API/docs/en-us/bulksheets/2-0/update-sp-campaigns#archive-a-campaign) explains how to archive a campaign--and this same logic would apply to any entity you want to archive. To archive, you only need to set the "Operation" to ARCHIVE, and be sure the entity ID is in the sheet. If you are doing this from a downloaded file, the ID will already be populated, so you literally just need to set the Operation, save the file, upload it, and you're done. 

**But are you saying that you tried putting DELETE into the "Operation" field, and that actually worked?** I don't think that should work, but want to be sure I understand what you are saying about that :)

**Reply by news2023** [reply]
Thank you so much for taking the time to answer all my questions, I really appreciate it. Upon reading the article on Amazon yes you're correct we only have 3 options under operations - create, update & archive. Thank you so much for confirming it.
